File: donghyo/Programmers/functionDevelop.py
from collections import deque

def solution(progresses, speeds):
    answer = list()
    subPercent = 0
    workDayList = deque()
    j = 0
    count = 1

    for i in range(len(progresses)):
        subPercent = 100 - progresses[i]
        workDay = 0
        full = 0

        while True:
            if subPercent > full:
                full += speeds[j]
                workDay += 1
            else:
                j += 1
                workDayList.append(workDay)
                break

    for i in range(1, len(workDayList)):
        if workDayList[i-count] < workDayList[i]:
            answer.append(count)
            count = 1
            
        else:
            count += 1

    answer.append(count)
    print(answer)
    return answer

solution([95, 90, 99, 99, 80, 99], [1, 1, 1, 1, 1, 1])  

File: donghyo/Programmers/FunctionDevelop.java
import java.util.ArrayList;
import java.util.Arrays;
import java.util.stream.Collector;
import java.util.stream.Collectors;

public class FunctionDevelop {
    public static void main(String[] args) {
        int[] progresses = {95, 90, 99, 99, 80, 99};
        int[] speeds = {1, 1, 1, 1, 1, 1};

        Solution call = new Solution();
        call.solution(progresses, speeds);
    }
}

class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        ArrayList<Integer> arr = new ArrayList<>();
        int[] answer = new int[progresses.length];
        int[] day = new int[progresses.length];
        int limit_percent = 100;
        int idx = 0;
        int answer_idx = 0;
        answer[0] = 1;

        while (idx <= progresses.length -1){
            int progresse = 0;
            int start = progresses[idx];

            while(progresse < limit_percent){
                progresse += start + speeds[idx];
                start = 0;
                day[idx]++;
            }
            if (day[idx] > 0 && idx >= 1){
                if(day[idx-1] >= day[idx]){
                    answer[answer_idx] += 1;

                }else{
                    answer_idx++;
                    answer[answer_idx] += 1;
                }     
                
            }
            idx++;
        }
        for(Integer e: answer){
            if (e != 0){
                arr.add(e);
            }
        }
        answer = arr.stream().mapToInt(i -> i).toArray();
        // System.out.println(Arrays.toString(answer));
        return answer;
    }
}
